### PR TITLE
handle non-routable pages (which return None as their url)

### PR DESCRIPTION
**Pull request by skirsdeda**
You get the idea: ```None + ''``` :)

### CODE REVIEW

**Comment by ababic** [comment]
Hey @skirsdeda. Is this really the desired behaviour when a page isn't routable? Is there anything more useful that could be returned? Or maybe just blank string? Also, I'm curious to know why you want to show non-routable pages in a menu and not just untick `show_in_menus` for those pages? Could you tell me a little bit more about your use case?

File: wagtailmenus/models/menuitems.py [hunk 1]
```diff
@@ -91,7 +91,11 @@ def menu_text(self):
 
     def relative_url(self, site=None):
         if self.link_page:
-            return self.link_page.relative_url(site) + self.url_append
```
Comment (ababic) [inline wagtailmenus/models/menuitems.py]: I'd be happier just changing this line to

`return self.link_page.relative_url(site) or '' + self.url_append`

**Comment by ababic** [comment]
@skirsdeda what I'm really looking to understand is, have you configured certain pages to be deliberately non-routable, or is there a configuration issue with your site that you're going to have to resolve somehow anyway (perhaps you've added pages at the same level as the homepage, when they should be below it?). 

Frankly I can't see any value in supporting non-routable pages in menus, since the pages themselves and their ancestors will likely also be non-routable, and so links aren't going to work. However, if it helps you achieve something deliberately that you couldn't otherwise achieve, I might accept a simpler change just to prevent the exception being thrown.

**Comment by skirsdeda** [comment]
@ababic I am working on translation support (single tree, no duplicate pages for different languages) for Wagtail (see https://github.com/skirsdeda/wagtail-translation). In current implementation any untranslated page (or page having untranslated page in it's path) is unroutable because it has no slug in selected language. And then Wagtail allows having unroutable pages anyway, so wagtailmenus should handle such situations.

**Comment by ababic** [comment]
@skirsdeda in that case, will you make the change to the code as requested?

**Comment by skirsdeda** [comment]
@ababic There's no point adding `self.url_append` to empty string. I changed the code to just return `''` if relative url is None.

**Comment by ababic** [comment]
@skirsdeda Let me be clear:

- I'm not interested in supporting non-routable pages in wagtailmenus 'out of the box'
- I'm won't add additional logic that might slow the rendering of all menus to support what I think is an incredibly rare edge-case

If you have custom needs that wagtailmenus doesn't cater for out of the box, you should look at [defining your own custom MenuItem model](https://github.com/rkhleics/wagtailmenus#12-overriding-the-default-wagtailmenus-models) and write your own method.

**Comment by skirsdeda** [comment]
@ababic Sure, whatever. As I said, there are many possibilities where you could get non-routable page. By not supporting this 'out of the box' you are leaving this edge-case as a bug for your end users. And I don't see how this would slow the rendering of all menus :)

**Comment by ababic** [comment]
@skirsdeda If enough people raise is as an issue, I will do something about it. Until now, you're the first, so it's not high priority.

**Pull request closed without merging**